auslib/db.py

```diff
@@ -1961,24 +1975,11 @@ def update(self, where, what, changed_by, old_data_version, transaction=None, dr
                 if not self.db.hasPermission(changed_by, "release", "modify", what["product"], transaction):
                     raise PermissionDeniedError("%s is not allowed to modify releases for product %s" % (changed_by, what["product"]))
 
-            # The way things stand right now we cannot grant access to _only_ modify
-            # the read only flag. When the permissions were still enforced at the
-            # web level we had this because that flag had its own endpoint.
-            # If we want this again we'll need to adjust this code, and perhaps
-            # make a special method on this class that only modifies read_only
-            # (similar to addLocaleToRelease).
-            if "read_only" in what:
-                product = what.get("product", current_release["product"])
-                # In addition to being able to modify the release overall, users
-                # need to be granted explicit access to manipulate the read_only
-                # flag. This lets us give out very granular access, which can be
-                # very helpful particularly in automation.
-                if what["read_only"] is False:
-                    if not self.db.hasPermission(changed_by, "release_read_only", "unset", product, transaction):
-                        raise PermissionDeniedError("%s is not allowed to mark %s products read write" % (changed_by, what.get("product")))
-                elif what["read_only"] is True:
-                    if not self.db.hasPermission(changed_by, "release_read_only", "set", product, transaction):
-                        raise PermissionDeniedError("%s is not allowed to mark %s products read only" % (changed_by, what.get("product")))
+            if is_readonly_change:
+                self.validate_readonly_change(
+                    where, what["read_only"], changed_by, release=current_release, transaction=transaction, dryrun=dryrun, signoffs=signoffs
+                )
+                continue
```

Comment:
This would read better as an if/else instead of using `continue`, since the remainder of the block below is relatively small.